pypeit/wavecalib.py

```diff
@@ -662,15 +662,33 @@ def build_wv_calib(self, arccen, method, skip_QA=False,
             # Now preferred
             if self.binspectral is None:
                 msgs.error("You must specify binspectral for the full_template method!")
-            final_fit = autoid.full_template(arccen, self.lamps, self.par, ok_mask_idx, self.det,
+            final_fit, order_vec = autoid.full_template(arccen, self.lamps, self.par, ok_mask_idx, self.det,
                                              self.binspectral, slit_ids=self.slits.slitord_id,
                                              measured_fwhms=self.measured_fwhms,
                                              nonlinear_counts=self.nonlinear_counts,
                                              nsnippet=self.par['nsnippet'])#,
                                              #debug=True, debug_reid=True, debug_xcorr=True)
+            # Grab arxiv for redo later?
+            if self.par['echelle']: 
+                # Hold for later usage
+                self.slits.ech_order = order_vec[:self.slits.nslits]
+                self.arccen = arccen
         elif self.par['method'] == 'echelle':
             # Echelle calibration files
-            angle_fits_file, composite_arc_file = self.spectrograph.get_echelle_angle_files()
+            try:
+                if self.spectrograph.multi_ech_settings:
+                    #print(self.par.keys())
+                    # kludge to handle Keck/NIRSPEC. Should potentially be implemented inside the NIRSPEC instrument stuff.
```

Comment:
Yes, I would:
- override the `get_echelle_angle_files` function in the Keck/NIRSPEC class
- add attributes to that class as necessary so that here, the only call is: `angle_fits_file, composite_arc_file = self.spectrograph.get_echelle_angle_files()`

Reply:
Agree!

Reply:
So far I've managed to cut the filter options but I'm working on getting the information about lamps passed into the NIRSPEC class.